### ISSUE TITLE
How much of this do you think we can get upstreamed?

### ISSUE THREAD

**Issue by Madouura** [nixified-ai/flake#48]
Hi there, was just curious if we could get most/all of this upstreamed to nixpkgs.
Haven't had a thorough look around yet, but at least a bunch of the python packages should be able to be upstreamed.
Also you might want to make sure things are working for ROCm 5.7.0. https://github.com/NixOS/nixpkgs/pull/258328 upgraded the ROCm stack.

**Comment by MatthewCroughan** [comment]
I have wanted to merge the overlays into Nixpkgs for quite some time now, but I lack the manpower, and I'm only one guy. I welcome it!

**Comment by Madouura** [comment]
For the stuff I do upstream from here, do you mind if I add you to the maintainers?

**Comment by MatthewCroughan** [comment]
@Madouura absolutely, and in fact can you please make a PR to remove it from here when you do? Or perhaps I can do that lazily as stuff gets into Nixpkgs, in one big PR.

**Comment by haras-unicorn** [comment]
@Madouura did u try doing this already? id like to contribute if i can!
if not ill make a draft pr by adding the python packages one by one and link the pr here
i think it would be easier to do it all in one big pr if nixpkgs maintainers are ok with that

**Comment by Madouura** [comment]
Currently side-tracked with another issue I'm trying to crack. If you have the time by all means go ahead, I'll try to review what I can.

**Comment by haras-unicorn** [comment]
alrighty then - mind u this would be my first nixpkgs contribution but hopefully i get some things right!
heres the [pr](https://github.com/NixOS/nixpkgs/pull/260968) - only added a single python package for now but will add more and a checklist of all things to do to get it merged
ill let u know when its ready for review or if i need assistance/guidance

**Comment by MatthewCroughan** [comment]
@haras-unicorn that's incredible, thank you so much, I hope Nixpkgs accepts it.

**Comment by haras-unicorn** [comment]
im kinda stuck on the dependency sets thing - any idea how to properly put that into inxpkgs?

**Comment by MatthewCroughan** [comment]
What is the dependency sets thing? I don't know exactly what you mean, happy to help if I can though!

**Comment by haras-unicorn** [comment]
ok looking at the code in the module folder the dependency-sets thing creates the dependency sets option and then the aipython3 module adds the "dependency sets" (all the packages we need through multiple overlays) through that which is then consumed in the koboldai and invokeai projects
i am just really unsure of how to add all this to nixpkgs since this is my first time contributing
ill ask some more on the nixos discord
